src/database_client.py

```diff
@@ -0,0 +1,117 @@
+        self._datasource = datasource
+        self._database_name = datasource["db"]
+        self._alternative_database = alternative_database
+        self._conn: connection = None
+
+    def _connect(self) -> None:
+        """Get connection.
+
+        Raises:
+            Error: something went wrong while connecting to the database.
+        """
+        if self._conn is None or self._conn.closed != 0:
+            logger.debug("Connecting to database")
+            try:
+                user = self._datasource["user"]
+                password = self._datasource["password"]
+                host = self._datasource["host"]
+                database_name = (
+                    self._alternative_database
+                    if self._alternative_database
+                    else self._datasource["db"]
+                )
+                self._conn = psycopg2.connect(
+                    f"dbname='{database_name}' user='{user}' host='{host}'"
+                    f" password='{password}' connect_timeout=5"
+                )
+                self._conn.autocommit = True
+            except psycopg2.Error as exc:
+                logger.exception("Failed to connect to database: %s", str(exc))
+                raise
+
+    def _close(self) -> None:
+        """Close database connection."""
+        if self._conn is not None:
+            self._conn.close()
+            self._conn = None
+
+    def prepare(self) -> None:
+        """Change database collate and ctype as required by Synapse.
+
+        Raises:
+            Error: something went wrong while preparing the database.
+        """
+        try:
+            self._connect()
+            with self._conn.cursor() as curs:
+                curs.execute(
+                    sql.SQL(
+                        "UPDATE pg_database SET datcollate='C', datctype='C' WHERE datname = {}"
+                    ).format(sql.Literal(self._database_name))
+                )
+        except psycopg2.Error as exc:
+            logger.error("Failed to prepare database: %s", str(exc))
+            raise
+        finally:
+            self._close()
+
+    def erase(self) -> None:
+        """Erase database.
+
+        Raises:
+            Error: something went wrong while erasing the database.
+        """
+        # Since is not possible to delete the database while connected to it
+        # this connection will use the template1 database, provided by PostgreSQL.
+        try:
+            self._connect()
+            with self._conn.cursor() as curs:
+                curs.execute(
+                    sql.SQL("DROP DATABASE {}").format(sql.Identifier(self._database_name))
+                )
+                curs.execute(
+                    sql.SQL(
+                        "CREATE DATABASE {} "
+                        "WITH LC_CTYPE = 'C' LC_COLLATE='C' TEMPLATE='template0';"
+                    ).format(sql.Identifier(self._database_name))
```

Comment:
Do you mean to use the format "cur.execute(sql_query, (user_input,))" instead?

This can be used too but I prefer to use the [module one](https://www.psycopg.org/docs/sql.html#module-usage) as is it used by the Data team [here](https://github.com/canonical/postgresql-k8s-operator/blob/main/lib/charms/postgresql_k8s/v0/postgresql.py#L114) too.

Reply:
Ok! Thanks for the explanation